node/resolvers/Mutations/Users.ts

```diff
@@ -509,6 +509,46 @@ const Users = {
       })
   },
 
+  createUserWithEmail: async (
+    _: void,
+    { orgId, costId, roleId, name, email, canImpersonate }: UserArgs,
+    ctx: Context
+  ) => {
+    const {
+      clients: { storefrontPermissions: storefrontPermissionsClient },
+      vtex: { logger },
+    } = ctx as any
+
+    try {
+      const result = await storefrontPermissionsClient.addUser({
+        orgId,
+        costId,
+        roleId,
+        name,
+        email,
+        canImpersonate,
+      })
+      return result.data.addUser
+    } catch (error) {
+      logger.error({
+        error,
+        message: 'addUser-error',
+      })
+      const message = error.graphQLErrors[0]?.message ?? error.message
```

Comment:
This `error.graphQLErrors` object may not be defined depending on the error, so you also need one more `?.` operator here:

```suggestion
      const message = error.graphQLErrors?.[0]?.message ?? error.message
```